src/doc_scan_service/session/create/session.specification.builder.js

```diff
@@ -154,6 +154,12 @@ class SessionSpecificationBuilder {
     return this;
   }
 
+  withSubject(subject) {
+    Validation.isPlainObject(subject, 'subject');
```

Comment:
Is there no validation on the object keys/values?

Reply:
That's done downstream, by DocScan API. 

We don't want to duplicate their logic, or take on their responsibility

Reply:
Why do we do any validation then? When we implement Typescript in the SDK, we'll have to be explicit about what the object's properties are. In that case developers using the SDK will have compile-time validation (and auto-complete for all config inputs), and so we could ditch any runtime validation done here and just leave that for docscan API.

Regardless, this seems to be the pattern for now, so it's fine.

Reply:
That is a good point, as typescript is really gonna move us towards being specific with the typing (keys and values), so either we just continue this approach of "loose" validation (only the top level type of what is being passed) or we become more granular (property names and value types).

Reply:
Small note: the addition of Typescript (TS) on top of this codebase does not make it mandatory to type the entire config data. **As a first step to such TS integration**, we would probably keep the same level of validation as we do now, without going into nested data shape.

If we ditch the run-time validation, would that mean that non-TS integrators would have no validation ??
If so, something like zod might make a lot of sense.

As a 2nd step to such TS integration, we can add more input config validation. For instance by relying on the Doc Scan API OpenAPI schema, so that we avoid duplication the logic in here.

Reply:
Yes, vanilla JS developers would not get the validation. I am in favour of having comprehensive validation at the SDK level regardless of TS/JS, so +1 for zod